### PR TITLE
Db setup

### PR DESCRIPTION
**Pull request by kimcarey**
- Implement endpoints for UserRegistration and UserLogin

### CODE REVIEW

**Review by KaminKevCrew — APPROVED** [review]
I think it looks good! Nice work.

**Review by CharlieLe3 — COMMENTED** [review]
Everything seems good to me. I do have a question though. Should we also do validation of the form in the back-end? like checking to see if the password is 6 in length, and if both the password and repeatPassword are both matching? or is that done when the request to register is made on a '/register' route?

- This is pertaining to the server/resources.py file UserRegistration class.

File: server/.env [hunk 1]
```diff
@@ -1,2 +1,3 @@
 TEAM_NAME="Shums, Kim, Kevin, Charlie"
-FLASK_ENV="development"
\ No newline at end of file
+FLASK_ENV="development"
+flask_app.config['SECRET_KEY'] = 'some-secret-string'
```
Comment (skassam21) [inline server/.env:3]: This should be:
SECRET_KEY='some-secret-string'

and in the code you would write:
flask_app.config['SECRET_KEY'] = config.SECRET_KEY

and in your config file, need to set SECRET_KEY to os.environ['SECRET_KEY']

File: server/models/users.py [hunk 2]
```diff
@@ -1,13 +1,67 @@
...
+        try:
+            num_rows_deleted = db.session.query(cls).delete()
+            db.session.commit()
+            return {'message': '{} rows deleted'.format(num_rows_deleted)}
+        except:
+            return {'message': 'Something went wrong'}
+
+    @staticmethod
+    def generate_hash(password):
+        return sha256.hash(password)
```
Comment (skassam21) [inline server/models/users.py:43]: sha256 is not a great password hashing function (https://stackoverflow.com/questions/11624372/best-practice-for-hashing-passwords-sha256-or-sha512)... we are also not using a salt

Can we use bcrypt and flask_bcrypt -  https://flask-bcrypt.readthedocs.io/en/latest/

File: server/resources.py [hunk 3]
```diff
@@ -1,34 +1,38 @@
...
 
 
 class UserLogin(Resource):
     def post(self):
-        data = parser.parse_args()
-        return data
+        data = user_schema.load(request.json)
+        current_user = UserModel.find_by_email(data['email'])
+        if not current_user:
+            return {'message': 'User {} doesn\'t exist'.format(data['email'])}
```
Comment (skassam21) [inline server/resources.py:30]: It's a security issue to tell the user that the user doesn't exist in logging in routes and can be exploited. Just always return Wrong credentials

File: server/resources.py [hunk 4]
```diff
@@ -44,10 +48,10 @@ def post(self):
 
 class AllUsers(Resource):
     def get(self):
-        return {'message': 'List of users'}
+        return UserModel.return_all()
 
     def delete(self):
-        return {'message': 'Delete all users'}
+        return UserModel.delete_all()
```
Comment (skassam21) [inline server/resources.py:54]: I don't know if we want these routes right now and exposed? Maybe for testing, but we should remove them if we deploy / go to production

Comment (kimcarey) [reply server/resources.py:54]: Correct, it's only for testing purposes, and will be removed prior to going into production

File: server/models/users.py (hunk 2 above)
Comment (kimcarey) [reply server/models/users.py:43]: noted - working on this now

**Pull request merged**